src/deploy/functions/backend.ts

```diff
@@ -427,7 +427,15 @@ async function loadExistingBackend(ctx: Context & PrivateContextFields): Promise
     return;
   }
 
-  const gcfV2Results = await gcfV2.listAllFunctions(ctx.projectId);
+  let gcfV2Results;
+  try {
+    gcfV2Results = await gcfV2.listAllFunctions(ctx.projectId);
+  } catch (err) {
+    if (err.status === 404 && err.message?.toLowerCase().includes("method not found")) {
```

Comment:
I thought this is the best place to make the check since the gcfV2 package shouldn't care what error is thrown.

Reply:
Isn't there another error thrown below (in `gcfV2Results.functions`) since `gcfV2Results` will be `undefined` in this case?

Reply:
(It obviously doesn't given the test cases, but I'm curious how it works 🤔 )

Reply:
I don't think I follow. If `gcfV2.listAllFunctions` throws an error we either return from the method or throw the error to the next method. Otherwise `gcfV2Results` is set to the internal `gcfV2` type of `ListFunctionsResponse` which has the `functions` member. 

I can initialize `gcfV2Results` to something like:
```
let gcfV2Results = {
    functions: [],
    unreachable: [],
  };
```
But I'm not sure that would really do much since TypeScript implicitly figures out this - `gcfV2Results: ListFunctionsResponse`

Reply:
Ah my mistake - I read `return` and thought `continue` 🤦 .